glossary/milestone.md

```diff
@@ -0,0 +1 @@
+Milestones are specific points in the development process of a [product](product.md) that have a [definition of done](../define-done/readme.md).
```

Comment:
I still don't know what a milestone is. *specific points* are too vague I think. They also have prioritisation and have a particular flow (inbox, commission, in production).